packages/appsync-modelgen-plugin/src/utils/process-belongs-to.ts

```diff
@@ -58,3 +53,15 @@ export function processBelongsToConnection(
     targetName: connectionFields[0] || makeConnectionAttributeName(model.name, field.name),
   };
 }
+
+export function getBelongsToConnectedField(field: CodeGenField, model: CodeGenModel, connectedModel: CodeGenModel): CodeGenField | undefined {
+  let otherSideDirectives = flattenFieldDirectives(connectedModel).filter(dir => {
+    const connectedField = connectedModel.fields.find(connField => { return connField.name === dir.fieldName; });
```

Comment:
we don't need to check for the `index` directive anymore?

Reply:
Not anymore, I wrote that for a specific case where it wasn't finding the correct connected V2 field but this more generic code ends up working fine with the unit tests I had in and is easier to follow